src/test/java/io/strimzi/kafka/bridge/http/HttpBridgeTest.java

```diff
@@ -3396,4 +3399,110 @@ void subscribeExclusiveTopicAndPattern(VertxTestContext context) throws Throwabl
         context.completeNow();
         assertTrue(context.awaitCompletion(TEST_TIMEOUT, TimeUnit.SECONDS));
     }
+
+    @Test
+    void consumerDeletedAfterInactivity(VertxTestContext context) throws InterruptedException, ExecutionException, TimeoutException {
+        String topic = "consumerDeletedAfterInactivity";
+        kafkaCluster.createTopic(topic, 1, 1);
+
+        String sentBody = "Simple message";
+
+        CompletableFuture<Boolean> produce = new CompletableFuture<>();
+        kafkaCluster.useTo().produceStrings(1, () -> produce.complete(true), () ->
+                new ProducerRecord<>(topic, 0, null, sentBody));
+        produce.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+
+        String name = "my-kafka-consumer";
+        String groupId = "my-group";
+
+        String baseUri = "http://" + BRIDGE_HOST + ":" + BRIDGE_PORT + "/consumers/" + groupId + "/instances/" + name;
+
+        JsonObject consumerJson = new JsonObject();
+        consumerJson.put("name", name);
+        consumerJson.put("format", "json");
+
+        CompletableFuture<Boolean> create = new CompletableFuture<>();
+        postRequest("/consumers/" + groupId)
+                .putHeader("Content-length", String.valueOf(consumerJson.toBuffer().length()))
+                .putHeader("Content-type", BridgeContentType.KAFKA_JSON)
+                .as(BodyCodec.jsonObject())
+                .sendJsonObject(consumerJson, ar -> {
+                    context.verify(() -> {
+                        assertTrue(ar.succeeded());
+                        HttpResponse<JsonObject> response = ar.result();
+                        assertEquals(HttpResponseStatus.OK.code(), response.statusCode());
+                        JsonObject bridgeResponse = response.body();
+                        String consumerInstanceId = bridgeResponse.getString("instance_id");
+                        String consumerBaseUri = bridgeResponse.getString("base_uri");
+                        assertEquals(name, consumerInstanceId);
+                        assertEquals(baseUri, consumerBaseUri);
+                    });
+
+                    create.complete(true);
+                });
+
+        create.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+
+        // After
+        Thread.sleep(timeout - 1000);
+
+        //////////////////
```

Comment:
where are you checking about the consumer deleted after 5 seconds? it's not clear to me, sorry :-(

Reply:
The test does this scenario
1. create consumer
2. wait a time < timeout
3. try to subscribe - this should be successful only if the consumer was found
4. wait a time > timeout
5. try to subscribe - this should fail because consumer was not found (it is deleted after timeout)

Reply:
so you are calling sleep in the main event loop so blocking the Vert.x event loop ... it's clear why you choose 5 secods as timeout because it's the timeout for Vert.x event loop as well so you wanted to avoid warning messages from Vert.x :-).
What I would do is ...

1. consumer creation
2. create a vert.x timer, if the creation was ok, running after the timeout
3. try to delete the consumer and check that it fails because there is no consumer with that name

wdyt?